test/unit/Twilio/Types/PriceUnitSpec.hs

```diff
@@ -0,0 +1,34 @@
+{-#LANGUAGE OverloadedStrings #-}
+{-#LANGUAGE InstanceSigs #-}
+module Twilio.Types.PriceUnitSpec where
+
+import Test.Hspec
+import Twilio.Types.PriceUnit
+import Data.Aeson
+import Test.QuickCheck hiding (Success)
+import Data.Text (Text, pack)
+
+main :: IO ()
+main = hspec spec
+
+spec :: Spec
+spec = do
+  describe "PriceUnit" $ do
+    describe "decoding from JSON" $ do
+      it "should work for USD" $ do
+        fromJSON (String "USD") `shouldBe` (Success USD)
+      it "should work for EUR" $ do
+        fromJSON (String "EUR") `shouldBe` (Success EUR)
+      it "should work for JPY" $ do
+        fromJSON (String "JPY") `shouldBe` (Success JPY)
+      it "should work for BTC" $ do
+        let expectedResults = (Success $ OtherPriceUnit "BTC")
+        fromJSON (String "BTC") `shouldBe` expectedResults
+      it "should work for arbitrary strings" $ property $
+        \code -> fromJSON (String code) == (Success $ OtherPriceUnit code)
```

Comment:
I guess `code` will be autogenerated? We may need to filter out "USD", "EUR", and "JPY" here.

Reply:
Hi. Yea, `code` is autogenerated. `QuickCheck` will auto-generate 100 random values of type `Text`. In light of that, I don't think the changes of it generating `"USD"`, `"EUR"`, or `"JPY"` are high. I can figure out how to filter out those values if you'd like, but I'm not sure we'd gain much. Let me know.

Reply:
I think it would be good to include. I think [`suchThat`](http://hackage.haskell.org/package/QuickCheck-2.10.1/docs/Test-QuickCheck.html#v:suchThat) could help here.

Reply:
Ah - thank you for pointing me in the right direction. I've updated the code based on what I saw in the [QuickCheck Manual](http://www.cse.chalmers.se/~rjmh/QuickCheck/manual.html) regarding Quantified Properties - I can't link directly to the section but it's obvious how to get there.

One downside to the current implementation is that if we add another data constant to the PriceUnit data type, we'd have to remember to update `notPriceUnitEnum`. I can live with that, but if you know of a best practice here, I'm always looking to improve.